tests/core/test_streams.py

```diff
@@ -78,9 +81,14 @@ class SimpleTestTap(Tap):
     """Test tap class."""
 
     name = "test-tap"
-    settings_jsonschema = PropertiesList(Property("start_date", DateTimeType)).to_dict()
+    config_jsonschema = PropertiesList(
+        Property("username", StringType, required=True),
+        Property("password", StringType, required=True),
+        Property("start_date", DateTimeType),
+        additional_properties=False,
```

Comment:
I think this fixes my use case. I'd almost be in favour of making the stricter behaviour the default though, as I find the current config validation a bit misleading: the tap says everything is fine, even though it isn't. Or should the `Config validation passed with 0 warnings.` message include a warning that it did not check for some types of errors, such as additional properties, unless this is set?
I'm conscious that this would be a "breaking change" in the sense that it would break apparently valid configs, but it would probably surface problems to tap users that they might not be aware of, which I think is a win. In my case, I was using `tap-github` with `skip_parent_stream` (with the final `s` missing), believing that I was skipping a whole bunch of API calls. This change would have made the problem more obvious, and easily fixed. Are there cases where the user would want to run with a "broken" config?

Reply:
@laurentS I'm also inclined to make this strict validation the default. A slightly more complicated approach (for us), that is backwards compatible, is to iterate over the individual validators and catch the one for `additionalProperties`. Wdyt?

Reply:
I don't really understand the difference between the 2 behaviours. But if there are more validators (there seems to be a long list in the code), is your second approach more flexible, possibly making it possible to perform more sorts of validations that would help prevent errors? If so, I'd lean towards that approach as a user of the sdk, but I also don't know how much extra work it would be to switch to that...